lib/pbench/server/api/resources/query_apis/search_api.py

```diff
@@ -0,0 +1,201 @@
+                    username within a given time range
+
+            "fields": List of run document fields end user is interested in. E.g.
+                    ["@timestamp", "run.controller", "run.name", ...]
+        """
+        user = json_data.get("user")
+        start = json_data.get("start")
+        end = json_data.get("end")
+        search_term = json_data.get("search_term")
+        # If the search_term paramter is not specified we will return all the elastisearch hits
+        if search_term is None:
+            search_term = ""
+        selected_fields = json_data.get("fields")
+        # If no fileds are specified, the query will return all the fields from elasticsearch hits
+        if selected_fields is None:
+            selected_fields = []
+
+        # We need to pass string dates as part of the Elasticsearch query; we
+        # use the unconverted strings passed by the caller rather than the
+        # adjusted and normalized datetime objects for this.
+        start_arg = f"{start:%Y-%m}"
+        end_arg = f"{end:%Y-%m}"
+
+        self.logger.info(
+            "Search query for user {}, prefix {}: ({} - {}) on query: {}",
+            user,
+            self.prefix,
+            start,
+            end,
+            search_term,
+        )
+
+        uri_fragment = self._gen_month_range("run", start, end)
+        self.logger.info("fragment, {}", uri_fragment)
+        return {
+            "path": f"/{uri_fragment}/_search",
+            "kwargs": {
+                "json": {
+                    "query": {
+                        "bool": {
+                            "filter": [
+                                {"term": self._get_user_term(user)},
+                                {
+                                    "range": {
+                                        "@timestamp": {"gte": start_arg, "lte": end_arg}
+                                    }
+                                },
+                            ],
+                            "must": {
+                                "query_string": {
+                                    "query": f"*{search_term}*",
+                                    "analyze_wildcard": "true",
+                                }
+                            },
+                        }
+                    },
+                    "sort": [
+                        {"@timestamp": {"order": "desc", "unmapped_type": "boolean"}}
+                    ],
+                    "_source": {"include": selected_fields},
+                },
+                "params": {"ignore_unavailable": "true"},
+            },
+        }
+
+    def postprocess(self, es_json: Dict[AnyStr, Any]) -> Dict[AnyStr, Any]:
+        """
+        Returns a summary of the returned Elasticsearch query results, showing
+        the list of dictionaries with user selected fields from request json as keys
+
+        [
+            {
+                "key": "alphaville.example.com",
```

Comment:
Unlike the dashboard code, you're not assigning a `"key"` ... I'm still not 100% sure whether that's good or bad; but if you're not doing it, you shouldn't document it.

Reply:
Added the key field.